File: frontend/src/services/ChatMessageService.js
import axios from "axios";

export default class ChatMessageService{
    getAll(){
        return axios.get("http://localhost:8080/api/chatMessage/getAll")
    }

    getById(id){
        return axios.get(`http://localhost:8080/api/chatMessage/getById?id=${id}`)
    }

    getByCustomerId(customerId){
        return axios.get(`http://localhost:8080/api/chatMessage/getByFkCustomerId?customerId=${customerId}`)
    }

    getByEmployeeId(employeeId){
        return axios.get(`http://localhost:8080/api/chatMessage/getByFkEmployeeId?employeeId=${employeeId}`)
    }

    getByCustomerIdAndEmployeeId(customerId, employeeId){
        return axios.get(`http://localhost:8080/api/chatMessage/getByFkCustomerIdAndFkEmployeeId?customerId=${customerId}&employeeId=${employeeId}`)
    }

    deleteById(id){
        return axios.delete(`http://localhost:8080/api/chatMessage/deleteById?id=${id}`)
    }

    add(values){
        return axios.post("http://localhost:8080/api/chatMessage/add", values)
    }


}

File: backend/src/main/java/ecrow/backend/business/abstracts/ChatMessageService.java
package ecrow.backend.business.abstracts;

import ecrow.backend.core.utilities.results.DataResult;
import ecrow.backend.core.utilities.results.Result;
import ecrow.backend.entities.concretes.ChatMessage;
import ecrow.backend.entities.dtos.ChatMessageAddDto;

import java.util.List;

public interface ChatMessageService {
    DataResult<List<ChatMessage>> getAll();
    DataResult<ChatMessage> getById(Integer id);
    DataResult<List<ChatMessage>> getByFkEmployeeId(Integer employeeId);
    DataResult<List<ChatMessage>> getByFkCustomerId(Integer customerId);
    DataResult<List<ChatMessage>> getByFkCustomerIdAndFkEmployeeId(Integer customerId, Integer employeeId);
    Result deleteById(Integer id);

    Result add(ChatMessageAddDto chatMessageAddDto);
}
